app/controllers/admin_incoming_message_controller.rb

```diff
@@ -36,6 +36,26 @@ def destroy
     redirect_to admin_request_url(@incoming_message.info_request)
   end
 
+  def bulk_destroy
+    if params[:commit] == "No"
+      redirect_to(admin_request_url(params[:request_id]))
+    end
+
+    @incoming_messages = IncomingMessage.where(:id => params[:ids].split(","))
+    if params[:commit] == "Yes"
+      info_request = InfoRequest.find(params[:request_id])
+      @incoming_messages.each do |message|
+        message.destroy
+        info_request.log_event("destroy_incoming",
+                               { :editor => admin_current_user,
+                                 :deleted_incoming_message_id => message.id })
+      end
+      info_request.expire
```

Comment:
Calling `#expire` looks like the right thing to do here, but its clearing the incoming message cached body without regenerating it:

![screen shot 2016-02-25 at 10 48 26](https://cloud.githubusercontent.com/assets/282788/13321254/e4e695b6-dbc5-11e5-920c-6cfcce91be84.png)


![screen shot 2016-02-25 at 10 48 57](https://cloud.githubusercontent.com/assets/282788/13321258/ebd28b46-dbc5-11e5-8376-dd07ea5695e5.png)